herbert_core/classes/instrument_classes/instrument/detectors/@IX_det_He3tube/replicate.m

```diff
@@ -1,23 +1,35 @@
 function obj_out = replicate (obj, n)
 % Replicate detector elements to make n-fold larger detector array
+% or reverse replication - get single object out of replicated one.
+% The first values will be used for reverse replication.
 %
-%   >> obj_out = reorder (obj, n)
+%   >> obj_out = replicate (obj, n)
 %
 % Input:
 % ------
 %   obj         Input object
 %   n           Number of times to replicate the detectors
+%               If n<0, replace replicated items with first element.
 %
 % Output:
 % -------
 %   obj_out     Output array such that all internal data has been
 %               repeated n times, increasing the nmber of detectors by a
-%               facator of n
+%               factor of n
+%               If n<1, reduce size to one item per property
 
 
 % Replicate the detector arrays
-obj_out = obj;
-obj_out.dia_    = repmat(obj.dia_,[n,1]);
-obj_out.height_ = repmat(obj.height_,[n,1]);
-obj_out.wall_   = repmat(obj.wall_,[n,1]);
-obj_out.atms_   = repmat(obj.atms_,[n,1]);
+if n<0
```

Comment:
Do you want this to happen for any negative number? e.g. `-1023`, or would it be best to have it only the case for `-1`? 

Not saying it should be either way, just want to think about unexpected behaviour.

Reply:
Thanks for noticing this one - the code does not make sense. The conditions on n in the header comments do not match. There is also no record of why this change has been made. As it i not expected that the n<0 condition will actually occur, allowing this trhough for the merge but will need further testing.